subsys/bluetooth/services/dis.c

```diff
@@ -45,105 +48,265 @@ static struct dis_pnp dis_pnp_id = {
 };
 #endif
 
+#if defined(CONFIG_BT_DIS_SYSTEM_ID)
+/*
+ * Casting to uint64_t since the value is at most 5 bytes, but will appear as a 32-bit literal if it
+ * is less, giving a warning when right-shifting by 32.
+ */
+static uint8_t dis_system_id[8] = {
+	BT_BYTES_LIST_BE24(CONFIG_BT_DIS_SYSTEM_ID_OUI),
+	BT_BYTES_LIST_BE40((uint64_t)CONFIG_BT_DIS_SYSTEM_ID_IDENTIFIER)};
+#endif
+
 #if defined(CONFIG_BT_DIS_SETTINGS)
-static uint8_t dis_model[CONFIG_BT_DIS_STR_MAX] = CONFIG_BT_DIS_MODEL;
-static uint8_t dis_manuf[CONFIG_BT_DIS_STR_MAX] = CONFIG_BT_DIS_MANUF;
+BUILD_ASSERT(sizeof(CONFIG_BT_DIS_MODEL) <= CONFIG_BT_DIS_STR_MAX + 1);
+BUILD_ASSERT(sizeof(CONFIG_BT_DIS_MANUF) <= CONFIG_BT_DIS_STR_MAX + 1);
+static uint8_t dis_model[CONFIG_BT_DIS_STR_MAX + 1] = CONFIG_BT_DIS_MODEL;
+static uint8_t dis_manuf[CONFIG_BT_DIS_STR_MAX + 1] = CONFIG_BT_DIS_MANUF;
```

Comment:
Alternatively
```suggestion
static uint8_t dis_model[CONFIG_BT_DIS_STR_MAX + 1] = CONFIG_BT_DIS_MODEL;
static uint8_t dis_manuf[CONFIG_BT_DIS_STR_MAX + 1] = CONFIG_BT_DIS_MANUF;
BUILD_ASSERT(sizeof(CONFIG_BT_DIS_MODEL) <= sizeof(dis_model);
BUILD_ASSERT(sizeof(CONFIG_BT_DIS_MANUF) <= sizeof(dis_manuf);
```

Reply:
Can be applied to all arrays I believe

Reply:
If this is done, then you will get both the BUILD_ASSERT failure and a warning of too wide literals.

Reply:
Ah, fair point.